astropy/visualization/lupton_rgb.py

```diff
@@ -241,22 +266,26 @@ class AsinhMapping(Mapping):
     def __init__(self, minimum, stretch, Q=8):
         Mapping.__init__(self, minimum)
 
-        epsilon = 1.0/2**23            # 32bit floating point machine epsilon; sys.float_info.epsilon is 64bit
+        epsilon = (
+            1.0 / 2**23
+        )  # 32bit floating point machine epsilon; sys.float_info.epsilon is 64bit
         if abs(Q) < epsilon:
             Q = 0.1
         else:
             Qmax = 1e10
             if Q > Qmax:
                 Q = Qmax
 
-        frac = 0.1                  # gradient estimated using frac*stretch is _slope
-        self._slope = frac*self._uint8Max/np.arcsinh(frac*Q)
+        frac = 0.1  # gradient estimated using frac*stretch is _slope
+        self._slope = frac * self._uint8Max / np.arcsinh(frac * Q)
 
-        self._soften = Q/float(stretch)
+        self._soften = Q / float(stretch)
 
     def map_intensity_to_uint8(self, I):
-        with np.errstate(invalid='ignore', divide='ignore'):  # n.b. np.where can't and doesn't short-circuit
-            return np.where(I <= 0, 0, np.arcsinh(I*self._soften)*self._slope/I)
+        with np.errstate(
+            invalid='ignore', divide='ignore'
+        ):  # n.b. np.where can't and doesn't short-circuit
```

Comment:
As above - wonder if we can identify thee cases automatically?

Reply:
Fixed by 5dded7a.